metadata/upload.go

```diff
@@ -296,45 +374,37 @@ func parseRows(
 					inputs = append(inputs, kInputs...)
 				}
 
-				files := []*BQInput{}
 				nonFiles := []*BQInput{}
-
-				sizedFiles := make(chan *BQInput, len(inputs))
-				errs := make(chan error, 1)
-
-				for _, input := range inputs {
-					if input.Type == "file" {
-						go func(file BQInput) {
-							f := &file.Value
-							size, err := getGcsSize(ctx, f.StringVal)
-							if err != nil {
-								if getStatus(err) == http.StatusForbidden {
-									fmt.Println(err)
-									f.Valid = false
-								} else if strings.Contains(err.Error(), "doesn't exist") {
-									f.Valid = false
-								} else {
-									errs <- fmt.Errorf("%s: %s", f.StringVal, err.Error())
-									return
-								}
-							}
-							f.StringVal = fmt.Sprintf("%f", size)
-							sizedFiles <- &file
-						}(*input)
+				// create & stuff a channel holding jobs
+				fileCnt := 0
+				var jobs = make(chan GcsQueryJob, len(inputs))
+				for i, input := range inputs {
+					if (input.Type == "file") {
+						fileCnt++
+						job := GcsQueryJob{i, input}
```

Comment:
is there a reason to use a separate `GcsQueryJob` class in this case? I think it might be easier to just pass it as `[]*BQInput` instead.

Reply:
I was experimenting while learning from examples, so this implementation has that imprinted.
The reason was to keep track of the ID of the `*BQInput` "to be blamed", were something to happen.